pkgs/development/embedded/openocd-rp2040/default.nix

```diff
@@ -0,0 +1,70 @@
+{ lib
+, stdenv
+, fetchgit
+, pkg-config
+, hidapi
+, libftdi1
+, libusb1
+, which
+, libtool
+, autoconf
+, automake
+, texinfo
+, git
+, libgpiod
+}:
+
+stdenv.mkDerivation {
+  pname = "openocd-rp2040";
+  version = "0.12.0";
+  src = fetchgit {
+    url = "https://github.com/raspberrypi/openocd";
+    rev = "4d87f6dcae77d3cbcd8ac3f7dc887adf46ffa504";
+    sha256 = "sha256-4pY4Kzad53mmnnJDLl2FBDuulUldIo1NcNYzZKSCZts=";
+  };
```

Comment:
```suggestion
  src = fetchgit {
    url = "https://github.com/raspberrypi/openocd";
    rev = "4d87f6dcae77d3cbcd8ac3f7dc887adf46ffa504";
    sha256 = "sha256-4pY4Kzad53mmnnJDLl2FBDuulUldIo1NcNYzZKSCZts=";
    fetchSubmodules = true
  };
```

Reply:
For more information look at the fetchGit section: https://nixos.org/manual/nixpkgs/stable/#fetchgit